velox/functions/FunctionRegistry.cpp

```diff
@@ -160,4 +160,79 @@ resolveVectorFunctionWithMetadata(
   return exec::resolveVectorFunctionWithMetadata(functionName, argTypes);
 }
 
+bool isCompanionFunctionName(
+    const std::string& name,
+    const std::unordered_map<std::string, exec::AggregateFunctionEntry>&
+        aggregateFunctions) {
+  static const std::vector<std::string> kCompanionFunctionSuffixList = {
+      "_partial", "_merge_extract", "_merge", "_extract"};
+  for (auto companionFunctionSuffix : kCompanionFunctionSuffixList) {
+    auto suffixOffset = name.rfind(companionFunctionSuffix);
+    if (suffixOffset != std::string::npos) {
+      return aggregateFunctions.count(name.substr(0, suffixOffset)) > 0;
+    }
+  }
+  return false;
+}
+
+std::vector<std::string> getScalarNames(const bool sortNames) {
```

Comment:
Remove const from parameter. Rename parameter to sortResult. Return value should be const.